### ISSUE TITLE
Navigation instructions are sometimes misleading or repetitive 

### ISSUE THREAD

**Issue by metbic** [osmandapp/OsmAnd#15170]
### Description
When driving straight ahead and approaching a crossroad with a right turn-lane, the voice tells me to keep left instead of telling me to drive straight ahead. In German language, this leads to most confusing, even dangerous, situations. 
When taking the right turn direction, the voice tells me to take a slight right turn "leicht rechts abbiegen" when there's only one choice of turning right. It never even tells me to take a sharp one.
On parking lots, with unclear paths or multiple paths possible, the voice will start to babble, telling me directions nonstop. 

### How to reproduce?
It's easy to reproduce as this happens all the time when driving around with a German voice used. 

### Your Environment
OsmAnd Version: 4.2.6
Android/iOS version: Android Pie
Device model: Sony

**Maps used (online or offline):**   offline, Germany, North Rhine Westphalia, District Arnsberg, 1.7.2022

**Comment by sonora** [comment]
In order for this to be actionable it seems mandatory that you specify one or several routes, stating what you receive vs. your expecration at esch point.

**Comment by sonora** [comment]
Put yourself in the shoes of a developer.:

> It's easy to reproduce as this happens all the time when driving around with a German voice used.

may just not be an option. 😉

You could select an illustrative route by specifying start and end points, then pointing out specific instructions you hear which you think are misleading. Or e.g. prepare a video recording and comment it. Just look at other Issues here reporting navigation flaws and how they do it.

Above all a specific reproduction will serve the purpose of verifying which app profile, routing engine and settings you use, if you are following a prepared track or not, all things essential to materialize any improvement.

**Comment by scaidermern** [comment]
Unfortunately the mobile version of github is repeatedly broken. Tell your browser to display the desktop version for uploading a file. Also I think you have to change the file extension since gpx is not in the list of accepted files to upload (can this be fixed for this project?).

**Comment by Zirochkabila** [comment]
> 

Tested on version 4.2.7, the language of the voice prompts German - could not be reproduced. Voice prompts work correctly.

Try to update the application, perhaps with an update, this problem will disappear.

If after the update voice prompts still do not work correctly - please specify:
1. Are you using TTS or Recorded voice prompts (if TTS then German or German (casual))?
2. Coordinates of the route on which you will hear a mistake in the voice prompts (great if the route is around 1 km - this will speed up checking)
3. If possible, a short video that will be heard and seen that the voice prompts are not true (you can use an application to record the screen, then you can fully concentrate on driving)
4. Log file: Menu > Help > Send logcat log

**Comment by sonora** [comment]
In addition,I have just checked that both German TTS voices use the concept of speaking "_Links halten_" for the 'bear left' turn case, while they speak "_Links einordnen_" for the 'keep left' (more lane assistance) case.

@metbic possibly applies a different interpretation of this when stating

>  this makes no sense, as 'links halten' only applies for situations with two straight ahead lanes

**Comment by scaidermern** [comment]
Please check if you have duplicated maps installed. E.g. a map of a larger region (a state, "Bundesland") and a map of a smaller region, e.g. a city, covering the _same area_. This could lead to such wrong voice instructions.

**Comment by xeruf** [comment]
>  On parking lots, with unclear paths or multiple paths possible, the voice will start to babble, telling me directions nonstop.

I have experienced this on various streets in Berlin, for example Groß-Berliner Damm. It told me to keep left, keep right etc. every few meters.
Maybe I can supply more details later.

**Comment by sonora** [comment]
There are two separate issues mentioned in this lengthy thread, maybe it would better result in something actionable if we broke this out into two separate issues and descriptions, each with precise examples of where they occur.

**Issue 1** is "_the voice tells me to take a slight right turn "leicht rechts abbiegen" when there's only one choice of turning right_ ",  seemingly assuming that OsmAnd announcing a "slight turn" means to discriminate between 2 different possible turn alternatives / forks, when in fact OsmAnd has no such logic yet but simply evaluates if your turn anlge is less than 45 degrees, or more than 135 degrees.

**Issue 2** is "_the voice will start to babble, telling me directions nonstop_", as far as I can see referring to too many announcements in too short succession, and/or maybe in combination with OsmAnd trying to assist with left/right lane selection while the resulting direction of travel is actually straight.

**Comment by AdamRGrey** [comment]
> **Issue 2** is "_the voice will start to babble, telling me directions nonstop_", as far as I can see referring to too many announcements in too short succession, and/or maybe in combination with OsmAnd trying to assist with left/right lane selection while the resulting direction of travel is actually straight.

I encounter this 2nd issue _a lot_, shall I post data here or make a new issue?

**Comment by DmitryAlexei** [comment]
@AdamRGrey you can post your issue here and we will try to reproduce it
we need start and end coordinates, export of your profile settings (menu > settings > your profile where issue can be observed > export profile. Please add .txt extension to the file

**Comment by salbr** [comment]
I have the same issue with the "keep left" instructions. You can reproduce it on this route: [https://osmand.net/map/navigate/?start=51.701458,7.949538&end=51.698982,7.957339&profile=car#16/51.701634/7.948165](https://osmand.net/map/navigate/?start=51.701458,7.949538&end=51.698982,7.957339&profile=car#16/51.701634/7.948165)

Here is my profile:

[Auto.osf.txt](https://github.com/user-attachments/files/24862364/Auto.osf.txt)

**Comment by hmartink** [comment]
This issue keeps my wife from using OSM~ currently. Before our next trip, I will check for overlapping maps (in fact, I recently removed some outdated ones). In the last years, this problem was less pressing, but this year (sorry, cannot be more specific) OSM~ became unusuable for us (for car navigation) because of this issue, especially the Issue 2 Sonora mentioned on December 26, 2022.

This is especially annoying on long roads (Autobahnen or long city roads) where you simply need to follow the road, so these "leicht links halten", "leicht rechts halten" etc. are really distracting (and dangerous, IMHO, because this might lead to unexpected lane changes, since the driver thinks that she has to leave the road very soon). I even experience the situation, where OSM~ was so busy explain us the "leicht links halten" etc., that we would have missed the actual turn, if I had not looked at the screen at that time.

This can be easily reproduced in Germany, so if you need more data I can recored it on our next trip (but we only go into unknown areas a few times a year). It would be *really* helpful if this issue (both 1 and especially 2) could be resolved. Currently, we can only use OSM~ if I turn the volume down and issue the instructions myself while my wife is driving.

If you need more data, simply tell me what exactly I should record on this trip, e.g. filming OSM~ from a second device?

**Comment by sonora** [comment]
Quite an old thread, was not aware it is still valid. 😉 I have just used the app's 'simulate' feature to navigate the route @salbr specified above. (Latest nightly v5.4.0, August North-Rhine-Westphalia map for Muenster, freshly downloaded). I get zero voice prompts between the '_Continue for 650m..._.' prompt near route start and '_You have reached your destination_'.

I use OsmAnd's voice navigation occasionally, and also on 'real' trips seem to not have heard any excessive prompting or strange '_Keep left/right_' on motorways in many years.

So something seems different on my setup, and on my side the current verdict is 'not reproducible'.  Meaning in order to investigate would probably need a lot more specific  information, like maybe specify a short route and then a location of a bogus prompt, probably together with app version, map versions and any specific app settings if you changed something from the app's default.

**Comment by hmartink** [comment]
I was not aware of the simulate feature, I will try this out in the next days. 

OsmAnd Version: 5.3
Android version: 14
Device model: Fairphone 4 under /e/
The maps are those currently delivered, where do I see their version?

Regarding the settings - please tell me which settings I should look up - we use OSM~ for quite a few years already, so I don't know which are the default settings anymore.

**Comment by sonora** [comment]
I'm not quite sure what to advise...

- Map versions (Edition dates) can be seen under _Menu > Maps and Resuroces > Local_ tab.
- To simulate arbitrary navigation, _temporarily_(!) enable the OsmAnd Develooment plugin. Then calculate a route from an arbitrary start point to and end point, and before starting the voice navigation, tap 'Settings' on the Route definition dialog and enable 'Simulation'.
- If you hear too many or lengthy voice prompts, you can try a less chatty voice, like in you car profile go to _Menu > Configure profile > Navigation settings > Voice prompts > Language (TTS)_, and select e.g. '**German (casual)**'. It mutes some prompt types, and shortens others.
- In your car profile, try  _Menu > Configure profile > Reset to default_
- Lastly: If you say you have been using OsmAnd for many years, and through many updates, on the same device, maybe something 'accumulated'. A legacy setting, download, etc., which causes this. In particular since I cannot see very many other users with similar complaints over time here. Do you have a second device to try a fresh installation? Or trust yourself to make a backup of all personal data like track recordings, favorites, etc., then perform a clean re-install?

Or you can wait for support to post instructions here on how to export relevant profile data with which they could try to reproduce the issue. I am just trying to help out, but cannot fully accompany the process.

**Comment by isso-git** [comment]
I think the difference is due to the "Announce" setting > "Turn-by-Turn directions" german "Schritt-für-Schritt..." setting in the following menu: 
https://osmand.net/docs/user/navigation/guidance/voice-navigation/#voice-settings 
I can hear the instruction "Keep to the left" in the nightly 5.4.0#8507 (2026-08-07), in german and english, the language setting has no effect.

**Comment by hmartink** [comment]
I just disabled "Turn-by-Turn" directions and will ask my wife to try it on her next trip (or our common trip). Then I will try the tips provided by Sonora. (But as we don't do navigated travel too often, it might take some time until I can report again).
